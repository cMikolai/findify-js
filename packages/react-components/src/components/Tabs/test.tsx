import React from 'react'
import { mount } from 'enzyme'
import { Tabs, Tab } from './index'

const tabData = [
  {
    header: (props) => <h1 {...props}>Hello world</h1>,
    disabled: false,
    children: <p>Wazzup!</p>
  },
  {
    header: (props) => <h1 {...props}>Goodbye world</h1>,
    disabled: true,
    children: <p>Wazzup, tab 2 here!</p>
  },
  {
    header: (props) => <h1 {...props}>See ya later, world</h1>,
    disabled: false,
    children: <p>Tab 3 it is!</p>
  }
]

describe('Tabs', () => {
  it('renders first tab by default', () => {
    expect(mount(createTabs(tabData))).toMatchSnapshot()
  })

  it('respects defaultSelectedIndex property', () => {
    expect(mount(createTabs(tabData, { defaultSelectedIndex: 2 }))).toMatchSnapshot()
  })


  it('respects selectedIndex property', () => {
    expect(mount(createTabs(tabData, { defaultSelectedIndex: 2, selectedIndex: 0 }))).toMatchSnapshot()
  })
})

function createTabs(tabArr, props = {}) {
  return (
    <Tabs {...props}>
      {
        tabArr.map(tabProps => React.createElement(Tab, tabProps))
      }
    </Tabs>
  )
}
