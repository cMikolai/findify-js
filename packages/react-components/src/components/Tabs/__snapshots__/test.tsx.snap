// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders first tab by default 1`] = `
<Tabs>
  <withStateHandlers(withProps(Component))
    theme={Object {}}
  >
    <withProps(Component)
      onTabClick={[Function]}
      selectedIndex={0}
      theme={Object {}}
    >
      <Component
        body={
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        }
        onTabClick={[Function]}
        selectedIndex={0}
        theme={Object {}}
      >
        <ul>
          <withHandlers(Component)
            active={true}
            disabled={false}
            header={[Function]}
            index={0}
            key="0/.0"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={true}
              disabled={false}
              header={[Function]}
              index={0}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={true}
            header={[Function]}
            index={1}
            key="1/.1"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={true}
              header={[Function]}
              index={1}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={false}
            header={[Function]}
            index={2}
            key="2/.2"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={false}
              header={[Function]}
              index={2}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
        </ul>
        <div>
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        </div>
      </Component>
    </withProps(Component)>
  </withStateHandlers(withProps(Component))>
</Tabs>
`;

exports[`Tabs respects defaultSelectedIndex property 1`] = `
<Tabs
  defaultSelectedIndex={2}
>
  <withStateHandlers(withProps(Component))
    defaultSelectedIndex={2}
    theme={Object {}}
  >
    <withProps(Component)
      defaultSelectedIndex={2}
      onTabClick={[Function]}
      selectedIndex={0}
      theme={Object {}}
    >
      <Component
        body={
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        }
        defaultSelectedIndex={2}
        onTabClick={[Function]}
        selectedIndex={0}
        theme={Object {}}
      >
        <ul>
          <withHandlers(Component)
            active={true}
            disabled={false}
            header={[Function]}
            index={0}
            key="0/.0"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={true}
              disabled={false}
              header={[Function]}
              index={0}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={true}
            header={[Function]}
            index={1}
            key="1/.1"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={true}
              header={[Function]}
              index={1}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={false}
            header={[Function]}
            index={2}
            key="2/.2"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={false}
              header={[Function]}
              index={2}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
        </ul>
        <div>
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        </div>
      </Component>
    </withProps(Component)>
  </withStateHandlers(withProps(Component))>
</Tabs>
`;

exports[`Tabs respects selectedIndex property 1`] = `
<Tabs
  defaultSelectedIndex={2}
  selectedIndex={0}
>
  <withStateHandlers(withProps(Component))
    defaultSelectedIndex={2}
    selectedIndex={0}
    theme={Object {}}
  >
    <withProps(Component)
      defaultSelectedIndex={2}
      onTabClick={[Function]}
      selectedIndex={0}
      theme={Object {}}
    >
      <Component
        body={
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        }
        defaultSelectedIndex={2}
        onTabClick={[Function]}
        selectedIndex={0}
        theme={Object {}}
      >
        <ul>
          <withHandlers(Component)
            active={true}
            disabled={false}
            header={[Function]}
            index={0}
            key="0/.0"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={true}
              disabled={false}
              header={[Function]}
              index={0}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={true}
            header={[Function]}
            index={1}
            key="1/.1"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={true}
              header={[Function]}
              index={1}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
          <withHandlers(Component)
            active={false}
            disabled={false}
            header={[Function]}
            index={2}
            key="2/.2"
            onClick={[Function]}
            theme={Object {}}
          >
            <Component
              active={false}
              disabled={false}
              header={[Function]}
              index={2}
              onClick={[Function]}
              theme={Object {}}
            >
              <li
                className=""
                onClick={[Function]}
              />
            </Component>
          </withHandlers(Component)>
        </ul>
        <div>
          <p
            changeTab={[Function]}
          >
            Wazzup!
          </p>
        </div>
      </Component>
    </withProps(Component)>
  </withStateHandlers(withProps(Component))>
</Tabs>
`;
